### PR TITLE
script to update path of sources loaded with remark-source to a versioned path

### PR DESCRIPTION
**Pull request by m-vdb**
**Proposed changes**:

When we update the `documentation` branch, we might be creating or editing the documentation for a specific minor version (e.g. `2.3.x`). Some of the doc files (`.mdx` files) contain codeblocks that are supposed to load files contained in the codebase thanks to `remark-source` (+ a script that copies the sources under the `docs/` folder during the `make prepare-docs` phase). Such instructions look like this:
```
```python (docs/sources/rasa/core/channels/rest.py)
```
What's problematic is that this will always load the file from the `main` version of the docs, not the version-specific doc, and **this is a bug**. One way of fixing it is to replace `docs/sources/` by `versioned_docs/version-2.3.x/sources/` (for version `2.3.x` in this case) for each version of the files.

Alternatively, I considered updating `remark-sources` plugin (with a fork), but this plugin doesn't have knowledge of the underlying file structure, only the content it's being passed for processing. So we're stuck with a custom step in our script.

I've run the changes I made in this PR in another branch created from `documentation` and pushed [this commit](https://github.com/RasaHQ/rasa/commit/9f0560ba2e98b0bb86ebcb561cce94a13a1a3983) that shows the result. After this branch is merged into `documentation`, I'll run the script directly on the branch and commit the changes.

💡 no need to update Rasa X it isn't using the `remark-source` plugin. We'll need to update Rasa SDK though.

**Status (please check what you already did)**:
- [ ] added some tests for the functionality
- [ ] updated the documentation
- [ ] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/main/changelog) for instructions)
- [ ] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

File: scripts/push_docs_to_branch.sh [hunk 1]
```diff
@@ -70,6 +70,12 @@ else
     fi
 fi
 
+CURRENTLY_EDITING_VERSION=${EXISTING_VERSION:-$NEW_VERSION}
+if [ ! -z "$CURRENTLY_EDITING_VERSION" ]
```
Comment (wochinge) [inline scripts/push_docs_to_branch.sh:74]:
```suggestion
if [ -n "$CURRENTLY_EDITING_VERSION" ]
```

File: docs/plugins/included_source.js [hunk 2]
```diff
@@ -69,11 +72,49 @@ async function getIncludedSources(options) {
...
-    // finally, write all the source files in the `sourceDir`
+    // finally, write all the source files in the source directory
     return await Promise.all(sourceFiles.map(async ([sourceFile, content]) => {
-        return await outputFile(`${sourceDir}/${sourceFile}`, content);
+        return await outputFile(`${docsDir}/${relativeSourceDir}/${sourceFile}`, content);
     }));
 };
 
 
+/**
```
Comment (wochinge) [inline docs/plugins/included_source.js:82]: Can we please add a short sentence about what this is doing?

**Review by wochinge — COMMENTED** [review]
I don't understand everything, but seemse to make sense. Thanks for the explanation and the example commit 🙌🏻 Should maybe Lu have another look?

File: docs/plugins/included_source.js (hunk 2 above)
Comment (m-vdb) [reply docs/plugins/included_source.js:82]: ah yeah, totally forgot about this, thanks for the heads up

**Review by lunelson — COMMENTED** [review]
Looks good. It's too bad there isn't a way to have the remark plugin be aware of the path of the file being transformed 🤷🏻 so they could use relative paths, or a way to specify a reference directory so they could just refer to `source/..` and it would resolve; it worries me a little bit to bind the content of all these docs to a specific file structure but it seems like the best way to go 👍🏻

**Review by lunelson — APPROVED** [review]

**Pull request merged**